uber/model_checks.py

```diff
@@ -119,6 +119,13 @@ def full_name(attendee):
 
 @validation.Attendee
 @ignore_unassigned_and_placeholders
+def legal_name(attendee):
+    if attendee.legal_name_opt and attendee.full_name == attendee.legal_name_opt:
+        return 'When entering a legal name, it must be different than your preferred name. Otherwise, leave it blank.'
```

Comment:
The reason I didn't do a presave_adjustment here is because of the odd user experience of if someone types in their same legal name and hits submit, suddenly their legal name dissapears (imagine trying to fill it in repeatedly and going "why won't it save, it keeps dissapearing").

The checkbox idea solves another one of my concerns that there's going to be a lot of confusion over what the heck legal name is and why it exists - I'm guessing many people have never been introduced to the concept. Putting it behind a checkbox whose description they're forced to read (to determine if they need to check it or not) will help a lot there. People who actually need separate names will probably already be thinking a lot about names when they're filling that section out and will more than likely notice and fill it out properly.